### PR TITLE
Refactor code (ES6)

### PR DESCRIPTION
**Pull request by tpazderka**

### CODE REVIEW

File: tests-js/fido2.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import startFido2 from '../src/fido2.js'
+import fetchMock from 'fetch-mock'
+import 'babel-polyfill'
+
+describe('Fido 2', () => {
+
+    test('general test', () => {
```
Comment (stinovlas) [inline tests-js/fido2.test.js]: This name is rather general (pun intended), I'd prefer something more specific.

File: tests-js/fido2.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import startFido2 from '../src/fido2.js'
+import fetchMock from 'fetch-mock'
+import 'babel-polyfill'
```
Comment (stinovlas) [inline tests-js/fido2.test.js]: Why is this import needed?

File: tests-js/fido2.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        )
+
+        document.getElementsByTagName('body')[0].appendChild(form)
+        document.getElementsByTagName('body')[0].appendChild(submit_button)
+
+        navigator.credentials = {get: jest.fn(public_key => {
+            expect(public_key).toMatchSnapshot()
+        })}
+
+        startFido2()
```
Comment (stinovlas) [inline tests-js/fido2.test.js]: The result of this call is not checked.

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const path = require('path')
+
+module.exports = {
+    entry: {
+        fido2: [
+            'babel-polyfill',
+            path.resolve(__dirname, 'src/fido2.js'),
```
Comment (stinovlas) [inline webpack.config.js]: I'd very much prefer having the JS in `assets` directory rather than `src` directory. There is a Python workflow and packaging convention, that puts all Python modules inside the `src` directory. And even if we don't follow that convention right now, we may decide to do so in the future. Therefore, using non-conflicting name seems to be a better future-proof solution.

File: django_fido/templates/django_fido/base.html [hunk 5]
```diff
@@ -26,5 +26,6 @@
                 {% block content %}{% endblock %}
             </div>
         </div>
+        <script type="text/javascript" src="{% url 'javascript-catalog' %}"></script>
```
[the remark this answers is not in the document]
Comment (ziima) [reply django_fido/templates/django_fido/base.html]: I'd move this to a form, where the JS which requires the catalog is also defined.

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,6300 @@
+{
```
Comment (ziima) [inline package-lock.json]: Is `package-lock.json` something that should be part of the repository? I don't think we add the lock file to repositories.

File: src/fido2.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
+const DJANGO_FIDO_FORM_ID = 'django-fido-form'
```
Comment (ziima) [inline src/fido2.js]: Can we move the JS source to `static` directory? I don't quite like the `src` directory, since it only contains the JS and not the `django_fido` itself.

File: tests-js/fido2.test.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    })
+
+})
```
Comment (ziima) [inline tests-js/fido2.test.js]: Check for registration should be also added.

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+const path = require('path')
```
Comment (ziima) [inline webpack.config.js]: We have to add the webpack to `setup.py`. But I can do that later in a separate issue.

File: tests-js/fido2.test.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import startFido2 from '../src/fido2.js'
```
Comment (ziima) [inline tests-js/fido2.test.js]: And we need to add the tests to travis.

File: package-lock.json (hunk 6 above)
Comment (stinovlas) [reply package-lock.json]: Generally yes, it's supposed to be kept in VCS. It's purpose is to lead to predictable environment during build and deploy (which also leads to the same transpiled code on different machines).

File: src/fido2.js (hunk 7 above)
Comment (stinovlas) [reply src/fido2.js]: See my discussion above. I proposed to move it to `assets/js` directory, the same location as in Ferda.

File: django_fido/templates/django_fido/base.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (stinovlas) [reply django_fido/templates/django_fido/base.html]: Still, after moving this line, its placed under `extrahead` tag, therefore below any other javascript that might need `gettext` function. Also, there is a comment from Vlasta.

File: .babelrc [hunk 11]
```diff
@@ -0,0 +1,4 @@
+{
+    "presets": ["@babel/preset-env"]
+  }
+
```
[the remark this answers is not in the document]
Comment (zelvik) [reply .babelrc]: Fixed

File: .gitignore [hunk 12]
```diff
@@ -101,3 +101,9 @@ ENV/
 
 # mypy
 .mypy_cache/
+
+# Ignore node_modules
+node_modules/
+
+# Ignore transpiled JS code
+django_fido/static/django_fido/js
```
[the remark this answers is not in the document]
Comment (zelvik) [reply .gitignore]: Fixed

File: django_fido/urls.py [hunk 13]
```diff
@@ -3,6 +3,8 @@
 
 from django.conf.urls import url
 from django.views.generic import TemplateView
+from django.urls import path
```
[the remark this answers is not in the document]
Comment (zelvik) [reply django_fido/urls.py]: Fixed

File: django_fido/urls.py [hunk 14]
```diff
@@ -15,4 +17,5 @@
         name='registration_done'),
     url('^authentication/$', Fido2AuthenticationView.as_view(), name='authentication'),
     url('^authentication/request/$', Fido2AuthenticationRequestView.as_view(), name='authentication_request'),
+    url('^jsi18n/$', JavaScriptCatalog.as_view(), name='javascript-catalog'),
```
[the remark this answers is not in the document]
Comment (zelvik) [reply django_fido/urls.py]: Fixed

File: package.json [hunk 15]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "django-fido",
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (zelvik) [reply package.json]: Fixed

File: package.json [hunk 16]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "django-fido",
```
[the remark this answers is not in the document]
Comment (zelvik) [reply package.json]: Fixed

File: django_fido/templates/django_fido/base.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (zelvik) [reply django_fido/templates/django_fido/base.html]: Fixed

File: tests-js/fido2.test.js (hunk 1 above)
Comment (zelvik) [reply tests-js/fido2.test.js]: Fixed

File: tests-js/fido2.test.js (hunk 2 above)
Comment (zelvik) [reply tests-js/fido2.test.js]: For async/await

File: tests-js/fido2.test.js (hunk 3 above)
Comment (zelvik) [reply tests-js/fido2.test.js]: Fixed

File: webpack.config.js (hunk 4 above)
Comment (zelvik) [reply webpack.config.js]: Fixed

File: tests-js/fido2.test.js (hunk 8 above)
Comment (zelvik) [reply tests-js/fido2.test.js]: Fixed

File: django_fido/templates/django_fido/base.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (stinovlas) [reply django_fido/templates/django_fido/base.html]: Vlasta suggested moving this link to `django_fido/forms.py`. I myself would prefer to leave it in base template, however I'd move it above the `extrahead` block, to load it before any other JavaScript on that page (the advantage is that we load the catalog only once, no matter how many other scripts use it). What you did is something in between, which doesn't make much sense.

[the remark this answers is not in the document]
Comment (zelvik) [reply django_fido/templates/django_fido/base.html]: Because I think same, I moved it back, before `extrahead`

File: .travis.yml [hunk 17]
```diff
@@ -1,14 +1,22 @@
...
-  - "3.7"
+matrix:
+    include:
+        - language: python
+          sudo: false
+          dist: xenial  # Avoid https://code.djangoproject.com/ticket/30055#comment:12
+          python:
+            - "3.5"
+            - "3.6"
+            - "3.7"
```
Comment (stinovlas) [inline .travis.yml]: After this change of `.travis.yml` Python 3.6 and 3.7 jobs are no longer run.

File: src/fido2.js (hunk 7 above)
Comment (stinovlas) [reply src/fido2.js]: We agreed on `assets/js` for JavaScript sources, `assets/css` for CSS sources and so on. The transpiled files are placed under the static files directory.

File: django_fido/templates/django_fido/base.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (stinovlas) [reply django_fido/templates/django_fido/base.html]: I'm OK with this change, but Vlasta should approve it as well.

[the remark this answers is not in the document]
Comment (ziima) [reply django_fido/templates/django_fido/base.html]: Yes, we can note that it is required step for translations in JS in README.

File: .travis.yml [hunk 18]
```diff
@@ -1,14 +1,35 @@
...
+      script:
+        - tox
+    - name: "Python 3.7"
+      language: python
+      python:
+        - "3.7"
+      install:
+        - pip install tox-travis
+      script:
+        - tox
```
Comment (ziima) [inline .travis.yml]: I'm trying to make it nicer in `travis-tests` branch.

Comment (ziima) [reply .travis.yml]: Looks like it works there.

File: tox.ini [hunk 19]
```diff
@@ -48,5 +51,7 @@ commands =
     # Check JS translations
     polint --show-msg django_fido/locale/cs/LC_MESSAGES/djangojs.po
     # Ensure catalog is complete - make C locales to generate POT files and compare it using the msgcmp
+    mkdir -p locale
     django-admin makemessages --locale C --no-obsolete --no-location --keep-pot --domain djangojs
-    msgcmp django_fido/locale/cs/LC_MESSAGES/djangojs.po django_fido/locale/djangojs.pot
+    msgcmp django_fido/locale/cs/LC_MESSAGES/djangojs.po locale/djangojs.pot
+    rm -r locale
```
[the remark this answers is not in the document]
Comment (zelvik) [reply tox.ini]: After previous discussion I moved assets to django_fido as we agreed all together.
Also, I fixed the test.

File: tox.ini [hunk 20]
```diff
@@ -49,4 +49,3 @@ commands =
     polint --show-msg django_fido/locale/cs/LC_MESSAGES/djangojs.po
     # Ensure catalog is complete - make C locales to generate POT files and compare it using the msgcmp
     django-admin makemessages --locale C --no-obsolete --no-location --keep-pot --domain djangojs
-    msgcmp django_fido/locale/cs/LC_MESSAGES/djangojs.po django_fido/locale/djangojs.pot
```
Comment (ziima) [inline tox.ini]: This shouldn't disappear.

File: .travis.yml (hunk 17 above)
Comment (zelvik) [reply .travis.yml]: This should be fied already.

File: tests-js/fido2.test.js (hunk 10 above)
Comment (zelvik) [reply tests-js/fido2.test.js]: Tests (js) are in travis (runnig after code style check)

File: src/fido2.js (hunk 7 above)
Comment (zelvik) [reply src/fido2.js]: This is not relevant, we agreed on move to django_fido/js and js-tests. This move is already done.

File: webpack.config.js (hunk 9 above)
Comment (zelvik) [reply webpack.config.js]: @ziima:
Could you do it, please? I am not absolutely sure what you mean.
Also, could you resolve unresolved discussions or add reply on them, please?
Thanks

File: tox.ini (hunk 20 above)
Comment (zelvik) [reply tox.ini]: Fixed
